R/oxcAARCalibratedDate.R

```diff
@@ -81,13 +81,138 @@ format.oxcAARCalibratedDate <- function(x, ...){
-  #     plotoxcAARDateGGPlot2(x, ...)
-  #   } else {
-  plotoxcAARDateSystemGraphics(x, ...)
-  # }
+plot.oxcAARCalibratedDate <- function(x, use_ggplot=T, ...){
+  if (requireNamespace("ggplot2", quietly = TRUE) & use_ggplot) {
+    plotoxcAARDateGGPlot2(x, ...)
+  } else {
+    plotoxcAARDateSystemGraphics(x, ...)
+  }
 }
+
+#' @importFrom "ggplot2" "ggplot" "aes" "geom_area" "labs" "theme_light" "geom_ribbon" "scale_y_continuous" "sec_axis" "ggplot_build" "geom_polygon" "scale_x_continuous" "annotate" "geom_errorbarh" "scale_alpha_manual"
+#' @importFrom "stats" "dnorm"
+plotoxcAARDateGGPlot2<-function(x, ...){
+  bc <- .data <- NULL
+  to_plot <- data.frame(dates=x$raw_probabilities$dates,
+                        probability = x$raw_probabilities$probabilities,
+                        class = "unmodelled")
+  if(!(is.null(x$posterior_probabilities)) & !(is.na(x$posterior_probabilities))) {
+    to_plot <- rbind(to_plot,
+                     data.frame(dates=x$posterior_probabilities$dates,
+                                probability = x$posterior_probabilities$probabilities,
+                                class = "modelled")
+                     )
+  }
+
+  cal_curve_df <- data.frame(bp = x$cal_curve$bp,
+                             bc = x$cal_curve$bc,
+                             sigma = x$cal_curve$sigma)
+
+  base_unit_y <- max(to_plot$probability)/25
+
+  cal_curve_df <- subset(cal_curve_df, bc >= min(to_plot$dates) & bc <= max(to_plot$dates))
+
+  cal_curve_df_old_min <- min(cal_curve_df$bp)
+  cal_curve_df_old_range <- diff(range(cal_curve_df$bp))
+  cal_curve_df_new_min <- max(to_plot$probability)
+  cal_curve_df_new_range <- diff(range(to_plot$probability))
+
+  this_bp_distribution<-NULL
+  this_bp_distribution$y <- pretty(c(x$bp+5*x$std, x$bp-5*x$std), n=20)
+  this_bp_distribution$x <- dnorm(this_bp_distribution$y,x$bp, x$std)
+  this_bp_distribution <- as.data.frame(this_bp_distribution)
+
+  this_bp_distribution$y_rescaled <- (this_bp_distribution$y - cal_curve_df_old_min) / cal_curve_df_old_range * cal_curve_df_new_range + cal_curve_df_new_min
+
+  cal_curve_df$bp_rescaled <-
+    (cal_curve_df$bp - cal_curve_df_old_min) / cal_curve_df_old_range * cal_curve_df_new_range + cal_curve_df_new_min
+
+  cal_curve_df$sigma_rescaled <-
+    cal_curve_df$sigma / cal_curve_df_old_range * cal_curve_df_new_range
+
+  m <- ggplot() + theme_light()
+
+  graph <- m +
+    geom_area(data = to_plot, aes(x=.data$dates,
+                                  y=.data$probability,
+                                  group = .data$class,
+                                  alpha = .data$class),
+              fill = "#fc8d62",
+              position = "identity",
+              color="#00000077"
+              ) +
+    labs(title = paste0(x$name,
+                        ": ",
+                        x$bp,
+                        "\u00B1",
+                        x$std),
+         caption = x$cal_curve$name,
```

Comment:
Maybe there is a way to add the more obvious "IntCal20" somewhere to make more clear, what "Atmospheric data from Reimer et al (2020)" means? I guess that's just what oxcal returns?

Reply:
Good point. Nevertheless, this takes the result directly from the oxcal response, I would rather keep this behavior?

Reply:
OK :shrug: